### ISSUE TITLE
[Form][2.8] Type "entity" not setting selected data when choice_value is a string attribute

### ISSUE THREAD

**Issue by rasanga** [symfony/symfony#17271]
I have the following code:

```
$builder->add('category', 'entity', [
                'required'      => false,
                'class'         => 'Ras\Bundle\DomainBundle\Entity\Category',
                'choice_value'  => 'name',
                'choice_label'  => 'label'
            ]);
```

When I select a category it does not give me the selected category entity, but when I changed ```choice_value``` to be ```id``` it worked fine. There seem to be an issue when the ``` choice_value ``` is a string attribute

NOTE:
```Ras\Bundle\DomainBundle\Entity\Category``` entity has following attributes: ```id, name, label```

**Comment by xabbuh** [comment]
Why do you want to change the `choice_value` property to `name`? Since the `name` property is apparently not the identifier of your `Category` entity, the entity type will not be able to find an entry for the submitted values.

**Comment by rasanga** [comment]
thank you for the quick response :+1: 

I thought it should be able to find entries by given ```choice_value``` field. can we add this as a feature I can make a PR?

**Comment by xabbuh** [comment]
What is the advantage of that instead of just using the id?

**Comment by rasanga** [comment]
When you use ```GET``` method to send form data it would great have to have URL query parameters in human readable manner

For example ```example.com/search?category=php``` is better than ```example.com/search?category=2```

See the difference?

**Comment by xabbuh** [comment]
I understand your use case, but I do not think that a choice or entity type is what you should use for that imo.

**Comment by Tobion** [comment]
The entity tyoe is using the primary key to look up the doctrine entity which of course won't find it with `name`. You can use a custom query builder for that.

**Comment by rasanga** [comment]
I used a query builder as follows but I had no luck:

```
$builder->add('category', 'entity', [
                'required'      => false,
                'class'         => 'Ras\Bundle\DomainBundle\Entity\Category',
                'choice_value'  => 'name',
                'choice_label'  => 'label',
                'query_builder' => function (CategoryRepository $repository) {
                    return $repository->createQueryBuilder('c')
                        ->addSelect('c.name')
                        ->addSelect('c.label')
                        ->groupBy('c.name')
                    ;
            ]);
```

**Comment by HeahDude** [comment]
This issue should be reopen until #17694 is merged. This is a bug of `EntityType` not handling `choice_value`.

**Comment by xabbuh** [comment]
Well, we can keep it closed as we already keep track of that bug in #13964 and #17693.

**Comment by igorpaladino** [comment]
Having this feature working would be a nice thing. I was dealing with the same issue. Meanwhile, a workaround solution would be using Data View Transformers http://symfony.com/doc/current/cookbook/form/data_transformers.html

**Comment by HeahDude** [comment]
@igorpaladino it fixed by #17694 and #17990 as of 2.7.11, 2.8.4 and 3.0.4. Note that using a custom `choice_value` will prevent an optimization of the loader, loading all the entities on submission instead of the submitted values only.

**Comment by empiricalx** [comment]
I have been going nuts trying to solved this problem, to find out is a bug, Using a the work around is just overkill, I solved the issue with jquery.
